clippy_lints/src/matches/match_as_ref.rs

```diff
@@ -47,15 +53,28 @@ pub(crate) fn check(cx: &LateContext<'_>, ex: &Expr<'_>, arms: &[Arm<'_>], expr:
             expr.span,
             format!("manual implementation of `Option::{method}`"),
             |diag| {
-                diag.span_suggestion_verbose(
-                    expr.span,
-                    format!("use `Option::{method}()` directly"),
-                    format!(
-                        "{}.{method}(){cast}",
-                        Sugg::hir_with_applicability(cx, ex, "_", &mut applicability).maybe_paren(),
-                    ),
-                    applicability,
-                );
+                if need_as_ref {
+                    diag.note("but the type is coerced to `&_`, and so `as_ref` will need to be used instead");
```

Comment:
I would prefer this message.

```suggestion
                    diag.note("but the type is coerced to a non-mutable reference, and so `as_ref` can be used instead");
```

Reply:
> I would prefer this message.

Yeah this was somewhat of a stand-in.

Though I'm not so sure about "can" here -- after all, it's not like `as_mut` would work by itself. This _could_ work if we first presented the `.as_mut().map()` option and only then proposed changing that to `.as_ref()` as a way to simplify the code, but we currently don't...

That said, there seems to be a general trend in lints to try to make the suggestion type-check by putting extra things on top of it (type ascriptions, casts, etc.), and those rarely -- if ever -- get acknowledged by the help message, not to mention getting the motivation behind them explained.

Reply:
The idea behind "can be used" is that it "should be used", but without telling the user that they did something wrong. We are showing them the possibility for doing so instead of `.as_mut().map()`.